src/pipdeptree/_discovery.py

```diff
@@ -1,26 +1,35 @@
 from __future__ import annotations
 
+import ast
 import site
+import subprocess  # noqa: S404
 import sys
 from importlib.metadata import Distribution, distributions
+from pathlib import Path
 from typing import Iterable, Tuple
 
 from packaging.utils import canonicalize_name
 
 
 def get_installed_distributions(
+    interpreter: str = str(sys.executable),
     local_only: bool = False,  # noqa: FBT001, FBT002
     user_only: bool = False,  # noqa: FBT001, FBT002
 ) -> list[Distribution]:
     # See https://docs.python.org/3/library/venv.html#how-venvs-work for more details.
     in_venv = sys.prefix != sys.base_prefix
     original_dists: Iterable[Distribution] = []
+    py_path = Path(interpreter).absolute()
 
     if local_only and in_venv:
         venv_site_packages = site.getsitepackages([sys.prefix])
         original_dists = distributions(path=venv_site_packages)
     elif user_only:
         original_dists = distributions(path=[site.getusersitepackages()])
+    elif py_path != Path(sys.executable).absolute():
+        cmd = [str(py_path), "-c", "import site; print(site.getsitepackages())"]
```

Comment:
Would it make sense to grab the potential package metadata from `sys.path`  instead? It looks like `importlib.metadata` uses it by [default](https://github.com/python/cpython/blob/f5bd65ed37bbccfa113438cf9e34feae6816ecb1/Lib/importlib/metadata.py#L366) if you never supply a `path` argument to `distributions()`.

I fetched your branch and I do get the expected results.

Reply:
I used `sitepackages` to keep the same as the code above. But you are right, it is more standard to use `sys` which `importlib.metadata` uses.

Reply:
Yes, and now that I've been thinking about it further, we have been using the `sys.path` provided by the custom Python environment before these changes (i.e. when doing just `pipdeptree --python path/to/python`). So I think we should do this 👍